### PR TITLE
Remove SecurityProfile

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-1559

### CODE REVIEW

**Comment by apmasell** [comment]
There's a failing test, but it's also failing on develop.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/store/ExperimentStore.java [hunk 1]
```diff
@@ -75,5 +76,5 @@
 
   Collection<Experiment> listByLibrary(long id) throws IOException;
 
-  Collection<Experiment> listByRun(long runId) throws IOException;
+  List<Experiment> listByRun(long runId) throws IOException;
```
Comment (avarsava) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/store/ExperimentStore.java]: Should these changes be in a different PR?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/ProjectImpl.java [hunk 2]
```diff
@@ -112,29 +109,15 @@
...
 
   @Column(nullable = false)
   @Temporal(TemporalType.TIMESTAMP)
   private Date lastUpdated;
 
   /**
-   * Construct a new Project with a default empty SecurityProfile
+   * Construct a new Project
    */
   public ProjectImpl() {
```
[the remark this answers is not in the document]
Comment (apmasell) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/ProjectImpl.java]: They are implied, but since the have (arguably useless) JavaDoc, I didn't want to delete them.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/store/ExperimentStore.java (hunk 1 above)
Comment (apmasell) [reply core/src/main/java/uk/ac/bbsrc/tgac/miso/core/store/ExperimentStore.java]: So, we actually use some kind of list for all storage, but return a mix of `Collection` and `List`. This mostly didn't matter because the security filters would was `<T> List<T> filterUnreadable(Collection<T>)`. With that gone, it was a more straight forward to update everything to use just `List`.

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java [hunk 3]
```diff
@@ -116,10 +111,4 @@ public void beforeDelete(PoolOrder object) throws IOException {
     poolService.save(pool);
   }
 
-  @Override
-  public void authorizeDeletion(PoolOrder object) throws IOException {
-    Pool pool = poolService.get(object.getPool().getId());
-    authorizationManager.throwIfNotWritable(pool);
```
Comment (djcooke) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java]: Removing this means that only admins will be able to delete orders. Please update user manual to indicate this

Comment (apmasell) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java]: Hmm. Should we replace that with a admin-or-creator?

Comment (djcooke) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java]: Probably. "Writeable" technically meant everyone before, but I don't think everyone should be able to delete orders

Comment (apmasell) [reply miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultPoolOrderService.java]: Sounds good to me.
